lib/data/external/GCP/GcpService.js

```diff
@@ -0,0 +1,199 @@
+const AWS = require('aws-sdk');
+const { errors } = require('arsenal');
+const Service = AWS.Service;
+
+const GcpSigner = require('./GcpSigner');
+
+AWS.apiLoader.services.gcp = {};
+const GCP = Service.defineService('gcp', ['2017-11-01']);
+Object.defineProperty(AWS.apiLoader.services.gcp, '2017-11-01', {
+    get: function get() {
+        const model = require('./gcp-2017-11-01.api.json');
+        return model;
+    },
+    enumerable: true,
+    configurable: true,
+});
+
+Object.assign(GCP.prototype, {
```

Comment:
Can you opt for class and extends instead of modifying the prototype directly?

Reply:
The defineService returns an object, so I had to modify the prototype directly.